### PR TITLE
Starfish/hotfix/full tx parameter problem

### PR DESCRIPTION
**Pull request by SzeChuanChilliSauce**
issue：#54 #55

### CODE REVIEW

File: rpc/eth/api.go [hunk 1]
```diff
@@ -104,11 +104,12 @@ func (api *PublicAPI) roundParamFromBlockNum(blockNum ethrpc.BlockNumber) (uint6
...
 }
 
-func (api *PublicAPI) getRPCBlockData(oasisBlock *block.Block) (uint64, ethtypes.Transactions, uint64, []*ethtypes.Log, error) {
+func (api *PublicAPI) getRPCBlockData(oasisBlock *block.Block, fullTx bool) (uint64, []interface{}, uint64, []*ethtypes.Log, error) {
 	encoded := oasisBlock.Header.EncodedHash()
 	bHash, _ := encoded.MarshalBinary()
 	blockNum := oasisBlock.Header.Round
-	ethTxs := ethtypes.Transactions{}
+	// ethTxs := ethtypes.Transactions{}
+	ethRPCTxs := []interface{}{}
```
Comment (ptrus) [inline rpc/eth/api.go:112]: Add a comment that `ethRPCTxs` will contain either `RPCTransactions` or the transaction hashes, depending on the `fullTx` parameter.

File: rpc/eth/api.go [hunk 2]
```diff
@@ -135,7 +136,21 @@ func (api *PublicAPI) getRPCBlockData(oasisBlock *block.Block) (uint64, ethtypes
 		}
 
 		gasUsed += uint64(ethTx.Gas())
-		ethTxs = append(ethTxs, ethTx)
+
+		if !fullTx {
+			hash := ethTx.Hash()
+			ethRPCTxs = append(ethRPCTxs, hash)
+		} else {
+			blockHash := common.HexToHash(oasisBlock.Header.EncodedHash().Hex())
```
Comment (ptrus) [inline rpc/eth/api.go:144]: Could `RPCTransaction` be loaded from the database here?
```
		dbTx, err := api.backend.QueryTransaction(ethTx.Hash())
		if err != nil {
			api.Logger.Error("Failed to QueryTransaction", "hash", ethTx.Hash(), "error", err.Error())
			continue
		}
		rpcTx, err := api.getRPCTransaction(dbTx)
		if err != nil {
			api.Logger.Error("Failed to getRPCTransaction", "hash", ethTx.Hash(), "error", err.Error())
			continue
		}
```

That way, you don't need the added `NewRPCTransaction2` code, which will in any case be only temporary as this whole part should be removed in https://github.com/starfishlabs/oasis-evm-web3-gateway/pull/44 (and replaced by loading everything from the db).

**Review by ptrus — COMMENTED** [review]
Left some minor comments

File: rpc/eth/api.go (hunk 2 above)
Comment (SzeChuanChilliSauce) [reply rpc/eth/api.go]: ok，We modified this code.

**Review by ptrus — APPROVED** [review]

**Comment by ptrus** [comment]
looks good!

**Pull request merged**